proxy/envoy/config.go

```diff
@@ -269,6 +270,22 @@ func buildOutboundHTTPRoutes(instances []*model.ServiceInstance, services []*mod
 					routes = append(routes, buildDefaultRoute(cluster))
 				}
 
+				if service.External {
+					for _, route := range routes {
+						route.HostRewrite = service.Hostname
```

Comment:
it was a design choice.  For example the proxy in the pod matches on this list of domains:
```
     "httpbin:80",
     "httpbin",
     "httpbin.default:80",
     "httpbin.default",
     "httpbin.default.svc:80",
     "httpbin.default.svc",
     "httpbin.default.svc.cluster:80",
     "httpbin.default.svc.cluster",
     "httpbin.default.svc.cluster.local:80",
     "httpbin.default.svc.cluster.local",
     "httpbin.org:80",
     "httpbin.org"
```
By setting route.HostRewrite to a known value, the egress proxy explicitly only listens/matches for one of those domains:
```
    "httpbin.org"
```